### DISCUSSION TITLE
xfce4 tray icon issue with gtk3 build

### DISCUSSION THREAD

**Discussion by dmillfree868** [audacious-media-player/audacious#990] [Ideas]
With the GTK3 build of audacious the tray icon gets corrupted if you set the size on 22 or 23. As you can see from the screenshot the icon gets distorted. I'm not sure if this is a baked in icon or if it's from the 96x96 svg file located in hicolor icons.

![Screenshot_2024-04-25_14-31-29](https://github.com/audacious-media-player/audacious/assets/95503349/23638443-ffca-4e25-a07a-0a9de5574b8e)

**Comment by radioactiveman** [comment]
To me it looks like a 24 px icon is used but does not fit the desired space.
We don't control this though, we only request the icon name(s), not the size or scaling.

~You may want to report this to the GTK developers. The status icon is deprecated though, I wouldn't expect a fix by them.~
Maybe the Xfce developers can also help here? They coincidentally [changed](https://gitlab.xfce.org/xfce/xfce4-panel/-/commit/309ffffb2fa1b94ffd2334a5a3861aa5906ebefb) the scaling behavior today...

https://github.com/audacious-media-player/audacious-plugins/blob/58cd04ff5bda3dff33f846a9dcb1ed5cebe29e55/src/statusicon/statusicon.cc#L83-L96

**Reply by jlindgren90** [reply]
The xfce change looks like it's for SNI (StatusNotifierItem) tray icons, not XEmbed which is used by GtkStatusIcon. With XEmbed, all drawing (and thus icon scaling also) is done on the client/application side.

I'm pretty sure this is a GtkStatusIcon bug; I have seen it before. It started with the GTK 2 -> 3 transition, if memory serves. And as you say, GtkStatusIcon is deprecated (because GTK is GNOME's toolkit now, and GNOME 3 does not support status icons). So this is unlikely to ever be fixed.
